### ISSUE TITLE
Deserialization fails when request is empty but a content-type header is present

### ISSUE THREAD

**Issue by CoryCharlton** [dj-nitehawk/FastEndpoints#147]
I recently updated from 4.0.0-beta1 to 4.2.0 and an endpoint where I use `EmptyRequest` began to fail with a `System.Text.Json.JsonException` (see details below). 

I believe this error is due to this [commit](https://github.com/dj-nitehawk/FastEndpoints/commit/88f387bcb0b52d7c3722d340e1a16532083a5fe7#diff-7c2dd294fcf5f2fe90dc8e0eb87220dab04c7b63a5cc534aa009720d8db45c1e). 

I'm working on creating a test to reproduce and then I'll re-add the content-length check.

Exception:

```
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCoreAsObject(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonConverter jsonConverter, Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ContinueDeserialize[TValue](ReadBufferState& bufferState, JsonReaderState& jsonReaderState, ReadStack& readStack, JsonConverter converter, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.ReadAllAsync[TValue](Stream utf8Json, JsonTypeInfo jsonTypeInfo, CancellationToken cancellationToken)
   at FastEndpoints.Endpoint`2.BindToModel(HttpContext ctx, List`1 failures, JsonSerializerContext serializerCtx, Boolean dontAutoBindForm, CancellationToken cancellation)
   at FastEndpoints.Endpoint`2.ExecAsync(HttpContext ctx, EndpointDefinition endpoint, CancellationToken cancellation)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

HEADERS
=======
Accept: */*
Connection: keep-alive
Host: localhost:7167
User-Agent: PostmanRuntime/7.29.0
Accept-Encoding: gzip, deflate, br
Cache-Control: no-cache
Content-Type: application/json
Postman-Token: d7a60dcd-a9a1-4884-bfcb-12a210f708b6
```

**Comment by dj-nitehawk** [comment]
we stopped relying on the `content-length` header because it was reported that some json clients do not set the content length reliably. infact was missing altogether in some cases. and we need to read/buffer the request body stream in order to check if there's data in the stream. this is not desirable from a performance standpoint.

so. if your client is setting "application/json" as the content-type, then there **must** be some valid json in the request body. even if it's just an empty json object `{}`. otherwise STJ will throw like you're seeing.

please see my replies in the following thread for a workaround: https://github.com/dj-nitehawk/FastEndpoints/issues/124

**Comment by CoryCharlton** [comment]
Sounds good. I came up with option 4 and am submitting a PR now. It's basically:

```
        if (isEmptyRequest)
        {
            req = new TRequest();
        } 
        else if (isPlainTextRequest)
        ...
```

**Comment by dj-nitehawk** [comment]
bloody hell mate!
the answer is that simple eh?
seriously...
well done!!!
will merge your PR asap...
🙏🙏🙏

**Comment by dj-nitehawk** [comment]
fix available in `v4.3.0-beta11`
thanks again @CoryCharlton

**Comment by ksydex** [comment]
Just got into this error and was really sad tryna find the solution for like 1h :(
